--- dwi-0.6.2/db_drivers/Makefile.am.orig	2007-06-13 19:12:50.000000000 +0200
+++ dwi-0.6.2/db_drivers/Makefile.am	2007-09-08 14:20:03.080896606 +0200
@@ -36,10 +36,7 @@ AM_CFLAGS = \
 libdwi_dbdrivers_la_SOURCES = \
 	dui-initdb.c 
 
-libdwi_dbdrivers_la_LDFLAGS = \
-	-module 
-
-libdwi_dbdrivers_la_LIBADD = 
+libdwi_dbdrivers_la_LIBADD = $(GLIB_LIBS) -ldl
 
 #-------------------------------------
 
@@ -52,7 +49,10 @@ libdwi_db_odbc_la_CFLAGS = \
 	${ODBC_CFLAGS}
 
 libdwi_db_odbc_la_LDFLAGS = \
-	-module \
+	-module -avoid-version
+
+libdwi_db_odbc_la_LIBADD = \
+	$(GLIB_LIBS) \
 	${ODBC_LIBS}
 
 #-------------------------------------
@@ -65,7 +65,10 @@ libdwi_db_libpg_la_CFLAGS = \
 	`pkg-config --cflags glib-2.0`  
 
 libdwi_db_libpg_la_LDFLAGS = \
-	-module \
+	-module -avoid-version
+
+libdwi_db_libpg_la_LIBADD = \
+	$(GLIB_LIBS) \
 	${PQ_LIBS}
 
 #-------------------------------------
@@ -78,7 +81,10 @@ libdwi_db_libdbi_la_CFLAGS = \
 	`pkg-config --cflags glib-2.0`  
 
 libdwi_db_libdbi_la_LDFLAGS = \
-	-module \
+	-module -avoid-version
+
+libdwi_db_libdbi_la_LIBADD = \
+	$(GLIB_LIBS) \
 	${DBI_LIBS}
 
 #-------------------------------------
--- dwi-0.6.2/dwi-gtk/Makefile.am.orig	2007-06-13 19:48:28.000000000 +0200
+++ dwi-0.6.2/dwi-gtk/Makefile.am	2007-09-08 14:34:41.030928127 +0200
@@ -41,7 +41,6 @@ noinst_HEADERS = \
 
 
 libdwi_gtk_la_LDFLAGS = \
-	-module \
 	`pkg-config --libs libxml-2.0 glib-2.0 gobject-2.0 gtk+-2.0 libgnomeui-2.0 libglade-2.0` 
 
 libdwi_gtk_la_LIBADD = 
--- dwi-0.6.2/src/Makefile.am.orig	2007-06-13 19:50:17.000000000 +0200
+++ dwi-0.6.2/src/Makefile.am	2007-09-08 14:35:16.308938506 +0200
@@ -65,8 +65,7 @@ noinst_HEADERS = \
 	util.h
 
 libdwi_la_LDFLAGS = \
-	-module \
 	`pkg-config --libs libxml-2.0 glib-2.0 gobject-2.0` 
 
-libdwi_la_LIBADD = 
-
+libdwi_la_LIBADD = \
+	../db_drivers/libdwi-dbdrivers.la
--- dwi-0.6.2/app/Makefile.am.orig	2007-06-13 20:36:44.000000000 +0200
+++ dwi-0.6.2/app/Makefile.am	2007-09-08 14:33:46.683831062 +0200
@@ -34,7 +34,6 @@ dwiinclude_HEADERS =   \
 	readfile.h
 
 libdwi_parse_la_LDFLAGS = \
-	-module \
 	`pkg-config --libs libxml-2.0 glib-2.0 gobject-2.0` 
 
 libdwi_parse_la_LIBADD = 
@@ -56,13 +55,17 @@ dwi_run_CFLAGS = \
 	`pkg-config --cflags libxml-2.0 glib-2.0 gobject-2.0 gtk+-2.0 libgnomeui-2.0 libglade-2.0` 
 
 dwi_run_LDFLAGS = \
-	${QOF_LIBS} \
-	${ODBC_LIBS} \
 	`pkg-config --libs libxml-2.0 glib-2.0 gobject-2.0 gtk+-2.0 libgnomeui-2.0 libglade-2.0` 
 
+if BUILD_QOF
+DWIQOF = ../dwi-qof/libdwi-qof.la
+endif
+
 dwi_run_LDADD = \
 	libdwi-parse.la \
 	../src/libdwi.la \
 	../dwi-gtk/libdwi-gtk.la \
-	../dwi-qof/libdwi-qof.la \
-	../db_drivers/libdwi-dbdrivers.la
+	$(DWIQOF) \
+	../db_drivers/libdwi-dbdrivers.la \
+	${QOF_LIBS} \
+	${ODBC_LIBS}
