--- dwi-0.6.2/db_drivers/Makefile.am.orig	2007-06-13 19:12:50.000000000 +0200
+++ dwi-0.6.2/db_drivers/Makefile.am	2007-09-08 14:20:03.080896606 +0200
@@ -36,10 +36,7 @@ AM_CFLAGS = \
 libdwi_dbdrivers_la_SOURCES = \
 	dui-initdb.c 
 
-libdwi_dbdrivers_la_LDFLAGS = \
-	-module 
-
-libdwi_dbdrivers_la_LIBADD = 
+libdwi_dbdrivers_la_LIBADD = $(GLIB_LIBS) -ldl
 
 #-------------------------------------
 
@@ -52,7 +49,10 @@ libdwi_db_odbc_la_CFLAGS = \
 	${ODBC_CFLAGS}
 
 libdwi_db_odbc_la_LDFLAGS = \
-	-module \
+	-module -avoid-version
+
+libdwi_db_odbc_la_LIBADD = \
+	$(GLIB_LIBS) \
 	${ODBC_LIBS}
 
 #-------------------------------------
@@ -65,7 +65,10 @@ libdwi_db_libpg_la_CFLAGS = \
 	`pkg-config --cflags glib-2.0`  
 
 libdwi_db_libpg_la_LDFLAGS = \
-	-module \
+	-module -avoid-version
+
+libdwi_db_libpg_la_LIBADD = \
+	$(GLIB_LIBS) \
 	${PQ_LIBS}
 
 #-------------------------------------
@@ -78,7 +81,10 @@ libdwi_db_libdbi_la_CFLAGS = \
 	`pkg-config --cflags glib-2.0`  
 
 libdwi_db_libdbi_la_LDFLAGS = \
-	-module \
+	-module -avoid-version
+
+libdwi_db_libdbi_la_LIBADD = \
+	$(GLIB_LIBS) \
 	${DBI_LIBS}
 
 #-------------------------------------
--- dwi-0.6.2/dwi-gtk/Makefile.am.orig	2007-06-13 19:48:28.000000000 +0200
+++ dwi-0.6.2/dwi-gtk/Makefile.am	2007-09-15 02:12:23.243090038 +0200
@@ -41,8 +41,14 @@ noinst_HEADERS = \
 
 
 libdwi_gtk_la_LDFLAGS = \
-	-module \
 	`pkg-config --libs libxml-2.0 glib-2.0 gobject-2.0 gtk+-2.0 libgnomeui-2.0 libglade-2.0` 
 
-libdwi_gtk_la_LIBADD = 
+libdwi_gtk_la_LIBADD = \
+	../app/libdwi-parse.la \
+	../src/libdwi.la
 
+../app/libdwi-parse.la:
+	$(MAKE) -C ../app libdwi-parse.la
+
+install-exec-local:
+	$(MAKE) -C ../app install-libLTLIBRARIES
--- dwi-0.6.2/src/Makefile.am.orig	2007-06-13 19:50:17.000000000 +0200
+++ dwi-0.6.2/src/Makefile.am	2007-09-08 14:35:16.308938506 +0200
@@ -65,8 +65,7 @@ noinst_HEADERS = \
 	util.h
 
 libdwi_la_LDFLAGS = \
-	-module \
 	`pkg-config --libs libxml-2.0 glib-2.0 gobject-2.0` 
 
-libdwi_la_LIBADD = 
-
+libdwi_la_LIBADD = \
+	../db_drivers/libdwi-dbdrivers.la
--- dwi-0.6.2/app/Makefile.am.orig	2007-06-13 20:36:44.000000000 +0200
+++ dwi-0.6.2/app/Makefile.am	2007-09-15 00:36:18.893399519 +0200
@@ -34,10 +34,9 @@ dwiinclude_HEADERS =   \
 	readfile.h
 
 libdwi_parse_la_LDFLAGS = \
-	-module \
 	`pkg-config --libs libxml-2.0 glib-2.0 gobject-2.0` 
 
-libdwi_parse_la_LIBADD = 
+libdwi_parse_la_LIBADD = ../src/libdwi.la
 
 
 # ---------------------------------
@@ -51,18 +50,15 @@ dwi_run_SOURCES = \
 dwi_run_CFLAGS = \
 	-I ../src \
 	-I ../dwi-gtk \
-	${QOF_CFLAGS} \
 	${ODBC_CFLAGS} \
 	`pkg-config --cflags libxml-2.0 glib-2.0 gobject-2.0 gtk+-2.0 libgnomeui-2.0 libglade-2.0` 
 
 dwi_run_LDFLAGS = \
-	${QOF_LIBS} \
-	${ODBC_LIBS} \
 	`pkg-config --libs libxml-2.0 glib-2.0 gobject-2.0 gtk+-2.0 libgnomeui-2.0 libglade-2.0` 
 
 dwi_run_LDADD = \
 	libdwi-parse.la \
 	../src/libdwi.la \
 	../dwi-gtk/libdwi-gtk.la \
-	../dwi-qof/libdwi-qof.la \
-	../db_drivers/libdwi-dbdrivers.la
+	../db_drivers/libdwi-dbdrivers.la \
+	${ODBC_LIBS}
--- dwi-0.6.2/dwi-qof/Makefile.am.orig	2007-06-14 18:37:38.000000000 +0200
+++ dwi-0.6.2/dwi-qof/Makefile.am	2007-09-15 00:34:43.440137667 +0200
@@ -34,7 +34,9 @@ dwiinclude_HEADERS = \
 	parse-qof.h
 
 libdwi_qof_la_LDFLAGS = \
-	-module              \
 	$(QOF_LIBS)          \
 	`pkg-config --libs glib-2.0` 
 
+libdwi_qof_la_LIBADD = \
+	../app/libdwi-parse.la \
+	../src/libdwi.la
--- dwi-0.6.2/Makefile.am.orig	2007-06-14 20:03:59.000000000 +0200
+++ dwi-0.6.2/Makefile.am	2007-09-15 02:13:32.831476707 +0200
@@ -11,7 +11,7 @@
 
 # Don't build qof subdirectories if QOF not installed
 
-SUBDIRS = db_drivers dwi dwi-gtk dwi-qof src glade app examples po
+SUBDIRS = db_drivers dwi src dwi-gtk dwi-qof glade app examples po
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = dwi-0.pc
